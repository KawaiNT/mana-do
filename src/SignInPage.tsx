import React, {useState} from 'react';

import {useHistory} from 'react-router-dom'
import Service from './service'

const SignInPage = () => {
    const [form, setForm] = useState({
        userId: '',
        password: ''
    });
    const history = useHistory();

    const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        const resp = await Service.signIn(form.userId, form.password)

        localStorage.setItem('token', resp)
        history.push('/todo')
    }

    const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.persist()
        setForm(prev=>({
            ...prev,
            [e.target.name]: e.target.value
        }))
    }

    return (
        <div className="login__container">
            <h2 style={{marginTop: 4}}>Login</h2>
            <form className="login-form" data-testid='login-form' onSubmit={signIn}>
                <label className="login-form__label" htmlFor="user_id">User id</label>
                <input
                    id="user_id"
                    name="userId"
                    required
                    placeholder='Enter user id'
                    data-testid="input-userId"
                    value={form.userId}
                    style={{marginBottom: 16}}
                    onChange={onChangeField}
                />
                <label className="login-form__label" htmlFor="password" >Password</label>

                <input
                    id="password"
                    name="password"
                    type="password"
                    required
                    placeholder='Enter password'
                    data-testid="input-password"
                    style={{marginBottom: 16}}
                    value={form.password}
                    onChange={onChangeField}
                />
                <button className="btn-submit" data-testid="btn-submit" type="submit" style={{marginTop: 4}}>
                    Sign in
                </button>
            </form>
        </div>
    );
};

export default SignInPage;