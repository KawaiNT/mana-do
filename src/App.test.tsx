import React from 'react';
import { render } from '@testing-library/react';
import SignInPage from './SignInPage';
import Service from './service'

describe('Login component tests', () => {
  test('Login Form render', () => {
    const { asFragment } = render(<SignInPage />)

    expect(asFragment(<SignInPage />)).toMatchSnapshot()
  })

  test('Login form should be in the document', () => {
    const { getByTestId } = render(<SignInPage />)
    const inputNode = getByTestId('login-form')

    expect(inputNode).toBeInTheDocument()
  })

  test('UserId field should be in the document', () => {
    const { getByTestId } = render(<SignInPage />)
    const userIdField = getByTestId('input-userId')

    expect(userIdField).toBeInTheDocument()
    expect(userIdField).toBeRequired()
  })

  test('Password field should be in the document', () => {
    const { getByTestId } = render(<SignInPage />)
    const passwordField = getByTestId('input-password')

    expect(passwordField).toBeInTheDocument()
    expect(passwordField).toBeRequired()
  })


  test('Submit button should be in the document', () => {
    const { getByTestId } = render(<SignInPage />)
    const btnSubmit = getByTestId('btn-submit')

    expect(btnSubmit).toBeInTheDocument()
    expect(btnSubmit).toHaveAttribute('type', 'submit')
  })
})
