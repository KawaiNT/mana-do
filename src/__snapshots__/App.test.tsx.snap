// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests Login Form render 1`] = `
<DocumentFragment>
  <div
    class="login__container"
  >
    <h2
      style="margin-top: 4px;"
    >
      Login
    </h2>
    <form
      class="login-form"
      data-testid="login-form"
    >
      <label
        class="login-form__label"
        for="user_id"
      >
        User id
      </label>
      <input
        data-testid="input-userId"
        id="user_id"
        name="userId"
        placeholder="Enter user id"
        required=""
        style="margin-bottom: 16px;"
        value=""
      />
      <label
        class="login-form__label"
        for="password"
      >
        Password
      </label>
      <input
        data-testid="input-password"
        id="password"
        name="password"
        placeholder="Enter password"
        required=""
        style="margin-bottom: 16px;"
        type="password"
        value=""
      />
      <button
        class="btn-submit"
        data-testid="btn-submit"
        style="margin-top: 4px;"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
